gooey/internal/scrollbar.lua

```diff
@@ -81,6 +82,49 @@ function M.vertical(handle_id, bounds_id, action_id, action, fn, refresh_fn)
 	return scrollbar
 end
 
+function M.horizontal(handle_id, bounds_id, action_id, action, fn, refresh_fn)
+	handle_id = core.to_hash(handle_id)
+	bounds_id = core.to_hash(bounds_id)
+	local handle = gui.get_node(handle_id)
+	local bounds = gui.get_node(bounds_id)
+	assert(handle)
+	assert(bounds)
+	local scrollbar = core.instance(handle_id, scrollbars, SCROLLBAR)
+	scrollbar.scroll = scrollbar.scroll or vmath.vector3(0, 0, 0)
+	scrollbar.vertical = false
+
+	local handle_size = gui.get_size(handle)
+	local bounds_size = gui.get_size(bounds)
+
+	scrollbar.enabled = core.is_enabled(handle)
+	scrollbar.node = handle
+	scrollbar.node_id = node_id
+	scrollbar.bounds = bounds
+	scrollbar.bounds_size = bounds_size
+	scrollbar.handle_size = handle_size
+
+	if action then
+		scrollbar.refresh_fn = refresh_fn
+
+		local action_pos = vmath.vector3(action.x, action.y, 0)
+
+		core.clickable(scrollbar, action_id, action)
+		if scrollbar.pressed_now or scrollbar.pressed then
+			local adjusted_x, adjusted_y = core.adjust(gui.get_adjust_mode(handle), action_pos.x, action_pos.y)
+			local bounds_pos = core.get_root_position(bounds)
+			local size = bounds_size.x - handle_size.x
+			local ratio = (size - (adjusted_x - bounds_pos.x - (scrollbar.handle_size.x / 2))) / size
+			SCROLLBAR.scroll_to(scrollbar, ratio, 0)
+			fn(scrollbar)
+		end
+	else
+		SCROLLBAR.scroll_to(scrollbar, scrollbar.scroll.x, 0)
+	end
+
+	scrollbar.refresh()
+	return scrollbar
+end
```

Comment:
*thought*: This part is mostly copy-pasted from `vertical`, which I'm not super happy about. However, I thought introducing abstractions to keep things DRY would make it less readable. I defer to you @britzl.

Reply:
I think this is perfectly fine to be honest. Sure, the first 20 lines or so could be moved to a shared function perhaps, but lets keep it as it is!